fix: Allow filtering null values

danieljbruce opened the pull request:
A user expressed interest in filtering with a null value. Currently this is possible if a null value is provided in a variable, but if a `null` literal is passed into `value` for `.filter` then we get a typescript compiler error. This typescript parameter change will now allow users to not experience the typescript compiler error when trying to filter by null.

Fixes #958

danieljbruce committed:
fix: Allows nulls to be passed into value

A user expressed interest in filtering with a null value. This typescript parameter change will now allow users to do so.

fixes #958

danieljbruce committed:
Add tests in the system test folder

The code in the system test folder reflects the user experience more closely so we want to provide test cases there that will break when compiling typescript if new changes in src are not provided.

kolea2 reviewed (commented):
LGTM, but please:

1. rename the PR to be clear this is for filtering (maybe something like 'fix: allow filtering null values')
2. Please have someone from yoshi-nodejs take a look

ruyadorno commented on system-test/datastore.ts:
```diff
@@ -868,6 +868,10 @@ describe('Datastore', () => {
 
+    it('should construct filters by null status', async () => {
+      datastore.createQuery('Character').filter('status', null);
+      datastore.createQuery('Character').filter('status', '=', null);
```
It would be better to assert the returning result of this filter operation in all tests.

danieljbruce committed:
Assert statements in system tests

Add assert statements to check that the value in the filter created in the system test actually equals null.

danieljbruce committed:
Modify tests to assert check for null

Slight rename in system tests. Modified tests so that they test to see that the filter has a null value.

danieljbruce committed:
Prefer elvis operator in assert statements

This change replaces the a longer fragment with elvis operator to make assert statements more concise.

danieljbruce committed:
eliminate the need for a variable

In two tests we eliminate variables and just have an assert statement so that we can have more concise code.

ruyadorno reviewed (approved):
LGTM 👍

danieljbruce committed:
Merge branch 'main' into nulls-allowed
